components/src/dynamo/common/backend/tests/test_sample_engine.py

```
@@ -164,6 +164,120 @@ async def test_decode_without_probe_marker_still_requires_prefill_result():
+    assert chunks[-1]["finish_reason"] == "length"
+
+
+async def test_downstream_rejects_malformed_encoder_result():
+    """Negative: downstream engine raises when encoder_result.handle has
+    an unexpected format, catching misconfigured encoder pipelines early."""
+    engine = SampleLLMEngine(
+        max_tokens=4,
+        delay=0.0,
+        disaggregation_mode=DisaggregationMode.AGGREGATED,
+    )
+    with pytest.raises(ValueError, match="encoder_result.handle"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)